code/modules/economy/Accounts_DB.dm

```diff
@@ -1,247 +1,177 @@
-				return
-			var/text
-			playsound(loc, 'sound/goonstation/machines/printer_thermal.ogg', 50, 1)
-			var/obj/item/paper/P = new(loc)
-			P.name = "financial account list"
-			text = {"
-				[accounting_letterhead("Financial Account List")]
-
-				<table>
-					<thead>
-						<tr>
-							<td>Account Number</td>
-							<td>Holder</td>
-							<td>Balance</td>
-							<td>Status</td>
-						</tr>
-					</thead>
-					<tbody>
-			"}
-
-			for(var/i in 1 to length(GLOB.all_money_accounts))
-				var/datum/money_account/D = GLOB.all_money_accounts[i]
-				text += {"
-						<tr>
-							<td>#[D.account_number]</td>
-							<td>[D.owner_name]</td>
-							<td>$[D.money]</td>
-							<td>[D.suspended ? "Suspended" : "Active"]</td>
-						</tr>
-				"}
-
-			text += {"
-					</tbody>
-				</table>
-			"}
-
-			P.info = text
-			visible_message("<span class='notice'>[src] prints out a report.</span>")
-			next_print = world.time + 30 SECONDS
-
-		if("print_account_details")
-			// Anti spam measures
-			if(next_print > world.time)
-				to_chat(usr, "<span class='warning'>The printer is busy spooling. It will be ready in [(next_print - world.time) / 10] seconds.")
-				return
-			var/text
-			playsound(loc, 'sound/goonstation/machines/printer_thermal.ogg', 50, 1)
-			var/obj/item/paper/P = new(loc)
-			P.name = "account #[detailed_account_view.account_number] details"
-			var/title = "Account #[detailed_account_view.account_number] Details"
-			text = {"
-				[accounting_letterhead(title)]
-				<u>Holder:</u> [detailed_account_view.owner_name]<br>
-				<u>Balance:</u> $[detailed_account_view.money]<br>
-				<u>Status:</u> [detailed_account_view.suspended ? "Suspended" : "Active"]<br>
-				<u>Transactions:</u> ([detailed_account_view.transaction_log.len])<br>
-				<table>
-					<thead>
-						<tr>
-							<td>Timestamp</td>
-							<td>Target</td>
-							<td>Reason</td>
-							<td>Value</td>
-							<td>Terminal</td>
-						</tr>
-					</thead>
-					<tbody>
-				"}
-
-			for(var/datum/transaction/T in detailed_account_view.transaction_log)
-				text += {"
-							<tr>
-								<td>[T.date] [T.time]</td>
-								<td>[T.target_name]</td>
-								<td>[T.purpose]</td>
-								<td>[T.amount]</td>
-								<td>[T.source_terminal]</td>
-							</tr>
-					"}
-
-			text += {"
-					</tbody>
-				</table>
-				"}
-
-			P.info = text
-			visible_message("<span class='notice'>[src] prints out a report.</span>")
-			next_print = world.time + 30 SECONDS
-
-#undef AUT_ACCLST
-#undef AUT_ACCINF
-#undef AUT_ACCNEW
+	. = account.deposit_credits(amount)
+	if(. && !supress_log)
+		var/database_log = amount >= DATABASE_LOG_THRESHHOLD ? TRUE : FALSE
+		log_account_action(account, amount, purpose, transactor, is_deposit = TRUE, log_on_database = database_log) //no if check here for now, since deposit credits currently will always return true
+	if(. && account == vendor_account)
```

Comment:
```suggestion
	. = account.deposit_credits(amount)
	if(!.)
		return

	if(!supress_log)
		var/database_log = amount >= DATABASE_LOG_THRESHHOLD ? TRUE : FALSE
		log_account_action(account, amount, purpose, transactor, is_deposit = TRUE, log_on_database = database_log) //no if check here for now, since deposit credits currently will always return true
	if(account == vendor_account)
```